common/persistence/nosql/nosqlplugin/cassandra/queue.go

```diff
@@ -188,7 +188,7 @@ func (db *cdb) InsertQueueMetadata(
 ) error {
 	clusterAckLevels := map[string]int64{}
 	query := db.session.Query(templateInsertQueueMetadataQuery, queueType, clusterAckLevels, version).WithContext(ctx)
-	_, err := query.ScanCAS()
+	_, err := query.ScanCAS(nil, nil, nil)
```

Comment:
I haven't researched this yet but it looks like Scylla returns all values for LWT while Cassandra returns only one. This code works for both, though.

Reply:
This might be answered by in [ScyllaDB docs](https://docs.scylladb.com/kb/lwt-differences/):
> Conditional statements return a result set, and unlike Cassandra, Scylla result set metadata doesn’t change from execution to execution: Scylla always returns the old version of the row, regardless of whether the condition is true or not. This ensures conditional statements work well with prepared statements.

Reply:
Thanks for contributing. 
This looks good to me, however, have you checked is there any other places need to change like this?

Reply:
I checked the source code and found one more place, however, it seems that that code path is not executed when running samples from https://github.com/uber-common/cadence-samples. It would panic otherwise. All other calls use `MapScanCAS()` which should be fine.

Reply:
Thanks. That's also the only place that I can see. 
The code path will be executed in XDC mode which is not used in sample by default.